Added a new chapter in auth (#7768)
island-is/island.is: handbook/technical-overview/auth/ids-admin.md

## Client

An application making protected resource requests on behalf of a resource owner (such as a user). The clients represent applications that can request tokens from the IAS.
### Creating clients
To create a client we need the following information: 
 - Your application domain (such as example.com)
 - Name of the application
	 - The name and domain are used to determine the client id. The name is also what users see when they are using the authentication service.
 - Callback URLs
	 - IAS compares these URLs with the callback URL sent with the request to the authorize endpoint. The callback URL sent with the request needs to match the URLs registered in IAS
 - Post logout redirect URLs
	 - These URLs are compared with the post-logout redirect URLs sent when ending a session. The URL in the request must match the URLs registered in IAS. The post logout redirect URL is the URL the user will be redirected to after signing out.
 - The application's technical stack
 - Whether the authentication is done client or server-side
 - If the stack supports PKCE

> 	 - PKCE stands for Proof key for code exchange. It is an extension of the authorization code flow to prevent CSRF and injection attacks.

Author: Ok, shouldn´t we remove it from the document as well?

> 	 - PKCE stands for Proof key for code exchange. It is an extension of the authorization code flow to prevent CSRF and injection attacks.

Reviewer: I would rephrase the question in the onboarding document. See comments on Slack about the onboarding document.

 - Whether you want national id (kennitala) and name in the id_token or fetch it through /connect/userinfo endpoint or some other way.
 - If you have any web services that need island.is authentication.
	 - If you need island.is authentication for your web services a resource and scopes are required.

> When we have this information we create the client and then give you the clientId and secret (the secret is not always used).

Author: Some clients, like native clients, do not have client secret.

> When we have this information we create the client and then give you the clientId and secret (the secret is not always used).

Reviewer: I would rephrase it to explicitly say that, for example:
```suggestion
When we have this information we create the client and then give you the clientId and secret (except when the client is used in native apps, then it does not use a client secret).
```

	 - Scopes are used to limit access to the resources such as API endpoints. An example of scope is a scope called calendar that represents access to calendar API. Scopes can be even more specific for example calendar.read and calendar.write where calendar.read is more restricted.
 - Resource
	 - A resource is something you want to protect with IAS (such as API) and has one or more scopes. Only one scope is needed when the API access is always the same. An example of that is the resource calendar with the scope calendar. An example of a resource with multiple scopes is a calendar resource with scopes calendar.read and calendar.write. These two scopes represent different levels of access to the calendar API and are grouped together by the resource.

### Creating resources and scopes

Usually, we create a client and then resources and scopes if needed and attach them to the client.
Sometimes you might only want resources and scopes created for you and don´t need a client. An example use case is when you want to use [island.is](http://island.is/) app to log in and when the user is logged in there they will be able to access your API. Regardless, the process is the same. We need:

>  - Name of the web service

Reviewer: ```suggestion
 - Domain, nationalId and email for the government agency responsible for the web service
 - Name of the web service
```

>  - Whether the web service is requires kennitolustyring

Reviewer: ```suggestion
 - Whether the web service is requires access control by nationalId (kennitölu stýringu)
```

 - If the scope should add national id (kennitala) and name of the user to the access_token

>  - If you have multiple access types for your web service you´ll need to provide them all with a description of what they do. We´ll also need your domain, national id (kennitala) and contact email.

Reviewer: ```suggestion
 - If you have multiple access levels for your web service, you´ll need to provide them all with a description of what they do.
```
